### DISCUSSION TITLE
VIA not working

### DISCUSSION THREAD

**Discussion by rawsp33d** [JellyTitan/Sofle-Pico#18] [Q&A] [answered]
Hey, I am writing this on my very own Sofle-Pico! I can't get VIA to work. Importing the json definition file works fine, but when I try to connect my device I get these two errors:

![image](https://github.com/JellyTitan/Sofle-Pico/assets/669012/446ac04b-291f-47f0-b543-be9857a8c812)

Any ideas? It shouldn't be an issue with browser permissions or feature flags because I have flashed microcontrollers from my browser before.

I can't screenshot it easily, but in my browser the "Ryan Neff sofle_pico" HID device does in fact pair and say "paired". But trying to connect to it through VIA is what gives the errors.

**Comment by rawsp33d** [comment, marked as answer]
Hmm I reflashed and it started working. I'm not sure what was wrong, as I had used the via uf2 files the first time around as well :shrug:. All is well in the world.

**Reply by JellyTitan** [reply]
Thanks for opening a ticket! I'll add a note about this to the troubleshooting guide. 

To the best of my knowledge, you're one of the first to build this board. Please let me know if there's anything that was unclear or could be improved in the build guide! (I'm working on refactoring the docs this week)!

**Comment by moraisz** [comment]
Hi, i known that topic is closed, but i have a tip about this problem to Linux users.

I reflashed many times and no work, until found this topic in [Arch Forum](https://bbs.archlinux.org/viewtopic.php?id=285709), basically you need change permission to browser write and read your keyboard.

You can solve this creating a udev rule:
```bash
sudo nano /etc/udev/rules.d/99-via.rules
```

In file add this line:
```bash
KERNEL=="hidraw*", SUBSYSTEM=="hidraw", MODE="0666"
```

Save, exit from nano and reload with:
```bash
sudo udevadm control --reload-rules
sudo udevadm trigger
```

Now try access VIA with Chromium based browser.

PS: sorry for my english, i try my better haha
